mllib-local/src/main/scala/org/apache/spark/ml/impl/Utils.scala

```diff
@@ -27,4 +27,55 @@ private[ml] object Utils {
     }
     eps
   }
+
+  /**
+   * Convert an n * (n + 1) / 2 dimension array representing the upper triangular part of a matrix
+   * into an n * n array representing the full symmetric matrix (column major).
+   *
+   * @param n The order of the n by n matrix.
+   * @param triangularValues The upper triangular part of the matrix packed in an array
+   *                         (column major).
+   * @return A dense matrix which represents the symmetric matrix in column major.
+   */
+  def unpackUpperTriangular(
```

Comment:
I thought breeze or commons math had a utility method for this already but I can't find it immediately

Reply:
OK, I will look for it.

Reply:
I can't find a packing or indexing method in scala or commons-math